From 2bed79a3794a0c0bd31ca2e3f72065aed4ea9876 Mon Sep 17 00:00:00 2001
From: me-cafebabe <me.cafebabe@gmail.com>
Date: Wed, 13 Sep 2023 16:22:42 +0800
Subject: [PATCH] partition: Enable Project ID Quota by default

* Reference: https://android.googlesource.com/platform/bootable/recovery/+/d21e516def1dde01bd3a5bcced9913442d88f6c5%5E%21/#F0
* While we're at it, Add it to EXT4 as well

Change-Id: I01e53d0e0329905949e3f26106ed8280e4781374
---
 partition.cpp | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/partition.cpp b/partition.cpp
index 18528f11..040e3849 100755
--- a/partition.cpp
+++ b/partition.cpp
@@ -2287,7 +2287,7 @@ bool TWPartition::Wipe_EXTFS(string File_System) {
 	gui_msg(Msg("formatting_using=Formatting {1} using {2}...")(Display_Name)("mke2fs"));
 
 	// Execute mke2fs to create empty ext4 filesystem
-	Command = "mke2fs -t " + File_System + " -b 4096 " + Actual_Block_Device + " " + size_str;
+	Command = "mke2fs -t " + File_System + " -b 4096 -I 512 " + Actual_Block_Device + " " + size_str;
 	LOGINFO("mke2fs command: %s\n", Command.c_str());
 	ret = TWFunc::Exec_Cmd(Command);
 	if (ret) {
@@ -2499,7 +2499,6 @@ bool TWPartition::Wipe_F2FS() {
 
 	bool NeedPreserveFooter = true;
 	bool needs_casefold = false;
-  	bool needs_projid = false;
 
 	Find_Actual_Block_Device();
 	if (!Is_Present) {
@@ -2509,7 +2508,6 @@ bool TWPartition::Wipe_F2FS() {
 	}
 
 	needs_casefold = android::base::GetBoolProperty("external_storage.casefold.enabled", false);
-	needs_projid = android::base::GetBoolProperty("external_storage.projid.enabled", false);
 	unsigned long long dev_sz = TWFunc::IOCTL_Get_Block_Size(Actual_Block_Device.c_str());
 	if (!dev_sz)
 		return false;
@@ -2518,8 +2516,9 @@ bool TWPartition::Wipe_F2FS() {
 		Length < 0 ? dev_sz += Length : dev_sz -= CRYPT_FOOTER_OFFSET;
 	char dev_sz_str[48];
 	sprintf(dev_sz_str, "%llu", (dev_sz / 4096));
-	if(needs_projid)
-		f2fs_command += " -O project_quota,extra_attr";
+
+	// Project ID
+	f2fs_command += " -O project_quota,extra_attr";
 
 	if(needs_casefold)
 		f2fs_command += " -O casefold -C utf8";
